File: 3217-delete-nodes-from-linked-list-present-in-array/3217-delete-nodes-from-linked-list-present-in-array.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {number[]} nums
 * @param {ListNode} head
 * @return {ListNode}
 */
var modifiedList = function(nums, head) {
      if (head==null)  return null;

      const hashset = new Set()
      for(let i=0;i<nums.length;i++){
        hashset.add(nums[i])
      }
      const dummyNode = new ListNode(-1);
      dummyNode.next = head;
      var current = dummyNode;
      while(current && current.next){
        if (hashset.has(current.next.val)){
            current.next = current.next.next;

        }
        else{
            current = current.next;
        }

      }

      return dummyNode.next

};

File: 3217-delete-nodes-from-linked-list-present-in-array/3217-delete-nodes-from-linked-list-present-in-array.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* modifiedList(vector<int>& nums, ListNode* head) {
        set<int>myset;
        for(int i=0;i<nums.size();i++){
            myset.insert(nums[i]);
        }
        ListNode* dummyNode = new ListNode(-1000);
        ListNode* current = dummyNode;
        while(head){
            if(myset.find(head->val)==myset.end()){
                current->next = head;
                current = current->next;

            }
            head = head->next;
           
        }
        current->next= NULL;
        return dummyNode->next;

        
    }
};